### PR TITLE
feat(nextjs-marketing-demo): add Header, Footer, and light mode styling [ALT-1270]

### PR DESCRIPTION
**Pull request by primeinteger**
## Purpose

For the Marketing Demo app, I added a Header, Footer, and light mode styling to align with the Figma design.

Screenshot 2024-09-25 at 3 53 10 PM

### CODE REVIEW

File: packages/templates/nextjs-marketing-demo/src/components/layout/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+'use client';
```
Comment (elylucas) [inline packages/templates/nextjs-marketing-demo/src/components/layout/Footer.tsx:1]: Is there a reason this and the header need to be client components?

File: packages/templates/nextjs-marketing-demo/src/components/layout/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+const Header: React.FC = () => {
+  const [current, setCurrent] = useState('home-page');
+
+  return (
+    <div className={styles.header}>
+      <div className={styles.logo}>
+        <span>Logo</span>
+      </div>
+      <div className={styles.nav}>
+        <Menu
```
Comment (elylucas) [inline packages/templates/nextjs-marketing-demo/src/components/layout/Header.tsx:23]: I can see theres some interactivity here, might be good to put Menu into its own component and mark that one as client so the Header itself can remain a RSC.

**Review by elylucas — APPROVED** [review]
Added a couple of nit pick comments on the use client components. Probably not too big of a deal so marking as approved.

File: packages/templates/nextjs-marketing-demo/src/components/layout/Header.tsx (hunk 2 above)
Comment (primeinteger) [reply packages/templates/nextjs-marketing-demo/src/components/layout/Header.tsx:23]: Good call, I'll separate the Menu into its own component.

File: packages/templates/nextjs-marketing-demo/src/components/layout/Footer.tsx (hunk 1 above)
Comment (primeinteger) [reply packages/templates/nextjs-marketing-demo/src/components/layout/Footer.tsx:1]: Definitely not needed in this one. It snuck in when I copy/pasted the Header component to create the Footer and I overlooked this line

**Pull request merged**